src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -0,0 +1,278 @@
+         */
+        this.category = '';
+
+        this.init(aircraftInitProps);
+    }
+
+    /**
+     *
+     *
+     * @property currentWaypoint
+     * @return {WaypointModel}
+     */
+    get currentWaypoint() {
+        return this.legCollection[0].currentWaypoint;
+    }
+
+    /**
+     *
+     *
+     * @property currentLeg
+     * @return {LegModel}
+     */
+    get currentLeg() {
+        return this.legCollection[0];
+    }
+
+    /**
+     *
+     * @property currentRoute
+     * @return {string}
+     */
+    get currentRoute() {
+        const routeSegments = _map(this.legCollection, (legModel) => legModel.routeString);
+
+        return routeSegments.join('..');
+    }
+
+    /**
+     *
+     *
+     * @for Fms
+     * @method init
+     * @param aircraftInitProps {object}
+     */
+    init(aircraftInitProps) {
+        this.category = aircraftInitProps.category;
+
+        this.legCollection = this._buildInitialLegCollection(aircraftInitProps);
+    }
+
+    /**
+     *
+     *
+     * @for LegModel
+     * @method destroy
+     */
+    destroy() {
+        this._navigationLibrary = null;
+        this._runway = '';
+        this.legCollection = [];
+        this.category = '';
+    }
+
+    /**
+     *
+     *
+     */
+    setModeControllerMode(modeSelector, mode) {
+        this._modeController[modeSelector] = mode;
+    }
+
+    /**
+     *
+     *
+     */
+    setModeControllerValue(fieldName, value) {
+        this._modeController[fieldName] = value;
+    }
+
+    updateModesForArrival() {
+        this._modeController.setForArrival();
+    }
+
+
+    updateModesForDeparture() {
+        this._modeController.setForDeparture();
+    }
+
+    /**
+     * Add a new `LegModel` to the left side of the `legCollection`
+     *
+     * @for Fms
+     * @method addLegToBeginning
+     * @param routeString
+     */
+    addLegToBeginning(routeString) {
+        const legModel = new LegModel(routeString, this._runway, this.category, this._navigationLibrary);
+
+        this.legCollection.unshift(legModel);
+    }
+
+    /**
+     * Move to the next possible waypoint
+     *
+     * This could be the next waypoint in the current leg,
+     * or the first waypoint in the next leg.
+     *
+     * @for LegModel
+     * @method nextWaypoint
+     */
+    nextWaypoint() {
+        if (!this.currentLeg.hasNextWaypoint()) {
+            this._moveToNextLeg();
+        }
+
+        this._moveToNextWaypointInLeg();
+    }
+
+
+    /**
+     * Given a `waypointName`, find where that waypoint exists within
+     * the `#legsColelction` then make that Leg active and `waypointName`
+     * the active waypoint for that Leg.
+     *
+     * @for Fms
+     * @method skipToWaypoint
+     * @param waypointName {string}
+     */
+    skipToWaypoint(waypointName) {
+        const { legIndex, waypointIndex } = this._findLegAndWaypointIndexForWaypointName(waypointName);
+
+        this.legCollection = _drop(this.legCollection, legIndex);
```

Comment:
oh, and to answer your question: we drop them because then moving through legs and waypoints is super clean.  the first item in waypoints and the first item in legs is assumed to _always_ be active.  we can always add a waypoint to the left or right of the list if we need to (not fully implemented yet).  doing it this way means we don't need to keep track of two sets of indicies; one for legs and one for waypoints.  Doing that requires an entire interface to manage moving them, resetting them, etc.